server/platform/services/sharedchannel/sync_send.go

```diff
@@ -405,6 +405,16 @@ func (scs *Service) processTask(task syncTask) error {
 		remotesMap[rc.RemoteId] = rc
 	}
 
+	// Filter out channel metadata system posts that don't need to be synchronized
+	if task.existingMsg != nil && task.existingMsg.Posts != nil {
+		task.existingMsg.Posts = filterChannelMetadataSystemPosts(task.existingMsg.Posts)
+	}
+
+	// Similarly for retry messages with posts
+	if task.retryMsg != nil && task.retryMsg.Posts != nil {
+		task.retryMsg.Posts = filterChannelMetadataSystemPosts(task.retryMsg.Posts)
+	}
```

Comment:
I'm thinking on moving this filter to the `filterPostsForSync` call that happens during the `syncForRemote` call. It would put the code side by side with the rest of the filtering and those posts would never reach the `retryMsg` property as we retry the payload after the filtering.

`existingMsg` is used only for status payloads, so it's not a concern at this point.

WDYT?

Reply:
On a second pass, I see that shared channels is the only caller for the `GetPostsSinceForSync`. What do you think about adding an `excludeChannelMetadataSystemPost` to the `GetPostsSinceForSyncOptions` and that way we can directly not fetch those posts from the DB?

I think it will simplify the code a lot, we could simply remove all the checks and filters as those posts would never be taken into account.

Reply:
Agree that moving it tot he DB makes more sense given the current codebase, since it appears all post fetching for sync goes through the `fetchPostsForSync` function. In the future , if there will be other codepaths to get to `GetPostsSinceForSyncOptions`, then we'd have to set the flag in those places as well. I updated the PR: